importer/src/path_canonicalizer.rs

```diff
@@ -628,22 +639,17 @@ fn check_machine(
                     check_expression(&module_location, e, state, &local_variables)
                 })?
             }
-            MachineStatement::LinkDeclaration(_, LinkDeclaration { flag, link, .. }) => {
-                check_expression(&module_location, flag, state, &local_variables)?;
-                link.params.inputs_and_outputs().try_for_each(|e| {
-                    check_expression(&module_location, e, state, &local_variables)
-                })?;
+            MachineStatement::LinkDeclaration(_, d) => {
+                for e in d.children() {
+                    check_expression(&module_location, e, state, &local_variables)?;
+                }
             }
             MachineStatement::InstructionDeclaration(_, _, instr) => {
-                for link_decl in &instr.links {
-                    check_expression(&module_location, &link_decl.flag, state, &local_variables)?;
-                    link_decl
-                        .link
-                        .params
-                        .inputs_and_outputs()
-                        .try_for_each(|e| {
-                            check_expression(&module_location, e, state, &local_variables)
-                        })?;
+                // Add the instruction parameters, ignore collisions.
```

Comment:
Shouldn't we only add the parameters that are actual parameters that introduce a new name i.e. only the typed ones?

Reply:
Good point, done